os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -74,42 +75,44 @@ typedef struct binary_header_s binary_header_t;
  *	 This function reads header and checks whether it is valid or not.
  *
  ****************************************************************************/
-static int binary_manager_read_header(int bin_idx, int part_idx, binary_header_t *header_data)
+static int binary_manager_read_header(char *path, binary_header_t *header_data)
 {
 	int fd;
 	int ret;
 	int read_size;
 	int file_size;
+	bool need_unlink;
 	uint32_t crc_hash = 0;
 	uint32_t check_crc = 0;
 	uint8_t crc_buffer[CRC_BUFFER_SIZE];
-	char devname[BINMGR_DEVNAME_LEN];
 
 	memset(header_data, 0, sizeof(binary_header_t));
 
-	snprintf(devname, BINMGR_DEVNAME_LEN, BINMGR_DEVNAME_FMT, BIN_PARTNUM(bin_idx, part_idx));
-	fd = open(devname, O_RDONLY);
+	need_unlink = false;
+
+	fd = open(path, O_RDONLY);
 	if (fd < 0) {
-		bmdbg("Failed to open %s: %d, errno %d\n", devname, ret, errno);
+		bmdbg("Failed to open %s: %d, errno %d\n", path, ret, errno);
 		return ERROR;
 	}
 
 	/* Read the checksum */
 	ret = read(fd, (FAR uint8_t *)&crc_hash, CHECKSUM_SIZE);
 	if (ret != CHECKSUM_SIZE) {
-		bmdbg("Failed to read %s: %d, errno %d\n", devname, ret, errno);
+		bmdbg("Failed to read %s: %d, errno %d\n", path, ret, errno);
 		goto errout_with_fd;
 	}
 
 	/* Read the binary header */
 	ret = read(fd, (FAR uint8_t *)header_data, sizeof(binary_header_t));
 	if (ret != sizeof(binary_header_t)) {
-		bmdbg("Failed to read %s: %d, errno %d\n", devname, ret, errno);
+		bmdbg("Failed to read %s: %d, errno %d\n", path, ret, errno);
```

Comment:
same comment as I wrote above. just read once.
I/O always takes much of time than any other operation.